pkg/http/server.go

```diff
@@ -0,0 +1,192 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package http
+
+import (
+	"crypto/tls"
+	"errors"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	humanize "github.com/dustin/go-humanize"
+)
+
+const (
+	serverShutdownPoll = 500 * time.Millisecond
+
+	// DefaultShutdownTimeout - default shutdown timeout used for graceful http server shutdown.
+	DefaultShutdownTimeout = 5 * time.Second
+
+	// DefaultTCPKeepAliveTimeout - default TCP keep alive timeout for accepted connection.
+	DefaultTCPKeepAliveTimeout = 10 * time.Second
+
+	// DefaultReadTimeout - default timout to read data from accepted connection.
+	DefaultReadTimeout = 30 * time.Second
+
+	// DefaultWriteTimeout - default timout to write data to accepted connection.
+	DefaultWriteTimeout = 30 * time.Second
+
+	// DefaultMaxHeaderBytes - default maximum HTTP header size in bytes.
+	DefaultMaxHeaderBytes = 1 * humanize.MiByte
+)
+
+// Server - extended http.Server supports multiple addresses and transparent redirection from insecure to secure connection.
```

Comment:
We don't have transparent redirection from insecure to secure connection anymore